api/api/domain/user.py

```diff
@@ -19,8 +27,15 @@ def get_validated_username(self):
         https://docs.aws.amazon.com/cognito/latest/developerguide/limits.html
         """
         if self.username is not None and re.fullmatch(USERNAME_REGEX, self.username):
-            return self.username
-        raise UserError("Invalid username provided")
+            custom_username_regex = os.environ.get("CUSTOM_USERNAME_REGEX")
+            if re.fullmatch(custom_username_regex, self.username):
```

Comment:
What happens if they don't specify a custom user name regex? We only want to perform a check if they have specified a custom regex